### PR TITLE
Make project key optional

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: neptune/management/internal/api.py [hunk 1]
```diff
@@ -115,9 +110,15 @@ def get_project_list(api_token: Optional[str] = None) -> List[str]:
 
 
 @with_api_exceptions_handler
+def _get_projects(params, api_token: Optional[str] = None) -> List:
```
Comment (Raalsky) [inline neptune/management/internal/api.py:113]: Let's structure it the same way as `_create_project` and `_get_workspace_id` (backend client as an input).

File: neptune/management/internal/api.py [hunk 2]
```diff
@@ -115,9 +110,15 @@ def get_project_list(api_token: Optional[str] = None) -> List[str]:
...
+    verify_type("api_token", api_token, (str, type(None)))
```
Comment (Raalsky) [inline neptune/management/internal/api.py:114]: I think we should verify as soon as possible (in user-exposed methods) and in private assume that parameters are correct

File: e2e_tests/management/test_management.py [hunk 3]
```diff
@@ -100,7 +99,7 @@ def test_standard_scenario(self, environment: Environment):
         assert project_identifier not in get_project_list(api_token=environment.admin_token)
 
     def test_visibility_workspace(self, environment: "Environment"):
-        project_name, project_key = a_project_name(project_slug=f"{fake.slug()}-workspace")
+        project_name, _ = a_project_name(project_slug=f"{fake.slug()}-workspace")
```
[the remark this answers is not in the document]
Comment (aniezurawski) [reply e2e_tests/management/test_management.py:102]:
```
    def test_service_accounts(self, environment: "Environment"):
        project_name, project_key = a_project_name(project_slug=f"{fake.slug()}-sa")
```
Are you sure it is merged?

[the remark this answers is not in the document]
Comment (Raalsky) [reply e2e_tests/management/test_management.py:102]: https://github.com/neptune-ai/neptune-client/blob/73b6cc6cfc299e67683f53934b758a341d5f5398/e2e_tests/management/test_management.py#L153

**Review by Raalsky — APPROVED** [review]

**Review by Raalsky — APPROVED** [review]

**Pull request merged**